crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -315,17 +323,29 @@ impl<'a, Provider: ResolverProvider, InstalledPackages: InstalledPackagesProvide
     }
 
     /// Resolve a set of requirements into a set of pinned versions.
-    pub async fn resolve(self) -> Result<ResolutionGraph, ResolveError> {
+    pub async fn resolve(mut self) -> Result<ResolutionGraph, ResolveError> {
         // A channel to fetch package metadata (e.g., given `flask`, fetch all versions) and version
         // metadata (e.g., given `flask==1.0.0`, fetch the metadata for that version).
         // Channel size is set large to accommodate batch prefetching.
-        let (request_sink, request_stream) = tokio::sync::mpsc::channel(300);
+        let (request_sink, request_stream) = tokio::sync::mpsc::unbounded_channel();
+        let solver_state = self.solver_state.take().unwrap();
 
         // Run the fetcher.
         let requests_fut = self.fetch(request_stream).fuse();
 
         // Run the solver.
-        let resolve_fut = self.solve(request_sink).boxed_local().fuse();
+        #[allow(unsafe_code)]
+        let solver: Solver<'static, InstalledPackages> =
+            unsafe { std::mem::transmute(Solver::new(&self, solver_state)) };
```

Comment:
This doesn't seem sound to me? Is there an alternative approach here that doesn't require `unsafe`? A `transmute`, in my experience, is typically quite tricky to get right. Especially when erasing away a lifetime.

I wonder if it makes sense to first focus on removing the lifetime from `Solver`/`Resolver`. I imagine that would simplify things here.

Reply:
Ah okay nice, that is exactly what you did in subsequent commits. Perfect. My bad for reviewing commit-by-commit. At a glance, it looked like that might be helpful given the size of the PR, but it ended up confusing me.

Reply:
Yeah, I initially did this just to test out the performance affects without having to make all the lifetime changes. Didn't mean for it to be merged in this state :sweat_smile: